### PR TITLE
PR and Main build github actions workflow (Re-Raised)

### PR DESCRIPTION
**Pull request by jaydee029**
Addresses isssue galasa-dev/projectmanagement#1875
(Re raised,  last [PR](https://github.com/galasa-dev/wrapping/pull/74) closed due to error )

### CODE REVIEW

**Comment by galasa-team** [comment]
Automatic triggering of the build is cancelled as user is not a member of the approved groups. Please ask an admin to review this PR and create a comment on the PR stating 'Approved for building'

**Comment by jaydee029** [comment]
#### comments from my side(last week)

The PR build is successful and running [here](https://github.com/jaydee029/wrapping/actions/runs/9627179547), while the Main build is repeatedly giving a gpg signing error.
image

The java setup action is usually sufficient for setting the gpg key and signing the artefcats as mentioned in the documentation.
Things that i have already tried-
- using a different action to setup the key, 
- have tried setting the key manually using terminal commands
- tried changing the maven sign plugin in the pom.xml file but that gave a forbidden 401 error.

Also i am able to sign and successfully upload the artefacts from my system using the same command.

To be noted the key is succesfully imported according to the java setup action
image

**Comment by jaydee029** [comment]
#### comments from Jade (last week)

Some comments so far - I believe you'll need to do something similar to the mavengpg Task which I've linked, which should resolve your GPG errors you are seeing with the Main build.

We can have a call if you need more discussion around this as I appreciate it looks a bit complex from just the files.

**Comment by jaydee029** [comment]
The recommended changes in the last [PR](https://github.com/galasa-dev/wrapping/pull/74) were fixed, however the error still persists.

**Comment by galasa-team** [comment]
Automatic triggering of the build is cancelled as user is not a member of the approved groups. Please ask an admin to review this PR and create a comment on the PR stating 'Approved for building'

File: .github/workflows/PR-build.yaml [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+            - name: Set up JDK 11
+              uses: actions/setup-java@v4
+              with:
+                java-version: '11'
+                distribution: 'temurin'
+                cache: maven
+            
+            - name: Building artefacts
+              run: |
+               mvn clean install -Dgpg.skip=true \
```
Comment (jadecarino) [inline .github/workflows/PR-build.yaml:26]: Nice, yes as you've provided the galasa.release.repo but for PRs we don't want to actually deploy there, `clean install` works better here

File: .github/workflows/build.yaml [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+
+
+          - name: Building artefacts
+            env:
+              GPG_PASSPHRASE: ${{secrets.GPG_PASSPHRASE}}
+              GPG_PRIVATE_KEY: ${{secrets.GPG_PRIVATE_KEY}}
+              GITHUB_ACTOR: ${{github.actor}}
+              GITHUB_TOKEN: ${{github.token}}
+            run: |
+             mvn -X --batch-mode -Dgpg.passphrase=GPG_PASSPHRASE deploy \
```
Comment (jadecarino) [inline .github/workflows/build.yaml:40]: What does `-X` mean?

File: pom.xml [hunk 3]
```diff
@@ -102,10 +104,7 @@
 							<goal>sign</goal>
 						</goals>
 						<configuration>
-							<gpgArguments>
-								<arg>--pinentry-mode</arg>
-								<arg>loopback</arg>
-							</gpgArguments>
+						    <defaultKeyring>false</defaultKeyring>
```
Comment (jadecarino) [inline pom.xml:107]: I think we want to add back in these lines as these just provided extra configuration to the GPG command-line tool, which I believe are still needed:
```
        <gpgArguments>
            <arg>--pinentry-mode</arg>
            <arg>loopback</arg>
        </gpgArguments>
```

Comment (jadecarino) [reply pom.xml:107]: Additionally, when disabling the default key ring, you need to provide the location of a custom key ring with it, like this:
```
<defaultKeyring>false</defaultKeyring>
    <keyring>${project.basedir}/path/to/custom/keyring.gpg</keyring>
```

Although can you please explain why we would not use the default keyring here?

**Review by jadecarino — CHANGES_REQUESTED** [review]

File: .github/workflows/build.yaml (hunk 2 above)
Comment (jaydee029) [reply .github/workflows/build.yaml:40]: It prints a more detailed trace of the build, we can remove this once errors are fixed.

**Comment by galasa-team** [comment]
Automatic triggering of the build is cancelled as user is not a member of the approved groups. Please ask an admin to review this PR and create a comment on the PR stating 'Approved for building'

**Comment by jadecarino** [comment]
Closing as wrapping workflows complete.

**Pull request closed without merging**